src/messages/MClientReply.h

```diff
@@ -390,13 +386,26 @@ class MClientReply final : public MMDSOp {
     using ceph::decode;
     auto p = payload.cbegin();
     decode(head, p);
+    // errorcode32_t implements conversion from/to different host error codes
+    // casting needed since error codes are signed int32 and head.result is unsigned int32
+    // errortype_t::code_t is alias for __s32, which is signed
+    // ceph_mds_reply_head::code_t is alias for __le32 which is unsigned
+    errorcode32_t temp;
+    temp.set_wire_to_host(static_cast<errorcode32_t::code_t>(head.result));
+    head.result = static_cast<ceph_mds_reply_head::code_t>(temp.code);
     decode(trace_bl, p);
     decode(extra_bl, p);
     decode(snapbl, p);
     ceph_assert(p.end());
   }
   void encode_payload(uint64_t features) override {
     using ceph::encode;
+    // errorcode32_t implements conversion from/to different host error codes
+    // casting needed since error codes are signed int32 and head.result is unsigned int32
+    // errortype_t::code_t is alias for __s32, which is signed
+    // ceph_mds_reply_head::code_t is alias for __le32 which is unsigned
+    errorcode32_t temp{static_cast<errorcode32_t::code_t>(head.result)};
+    head.result = static_cast<ceph_mds_reply_head::code_t>(temp.get_host_to_wire());
```

Comment:
Actually this won't work because it'll no longer be idempotent: the Messenger layer expects to be able to call `encode_payload` multiple times on message retries. You need to  make a copy of the `head` struct and change the copy's `result` field.